tiledb/api/src/group/mod.rs

```diff
@@ -0,0 +1,745 @@
+                "cols",
+                Datatype::Int32,
+                &[1, 4],
+                &4,
+            )?
+            .build();
+
+            DomainBuilder::new(&tdb)?
+                .add_dimension(rows)?
+                .add_dimension(cols)?
+                .build()
+        };
+
+        // Create a single attribute "a" so each (i,j) cell can store an integer
+        let attribute_a =
+            AttributeBuilder::new(&tdb, "a", Datatype::Int32)?.build();
+
+        // Create array schema
+        let schema = SchemaBuilder::new(&tdb, array_type, domain)?
+            .tile_order(TileOrder::RowMajor)?
+            .cell_order(CellOrder::RowMajor)?
+            .add_attribute(attribute_a)?
+            .build()?;
+
+        // Create array
+        Array::create(&tdb, array_uri, schema)?;
+        Ok(())
+    }
+
+    #[test]
+    fn test_group_functionality() -> TileDBResult<()> {
+        let tmp_dir =
+            TempDir::new().map_err(|e| Error::Other(e.to_string()))?;
+
+        let tdb = Context::new()?;
+        let group_path = tmp_dir.path().join("group2");
+        let group_uri = group_path.to_str().unwrap();
+        Group::create(&tdb, group_uri)?;
+
+        let group_read =
+            Group::open(&tdb, group_uri.to_owned(), QueryType::Read)?;
+        let group_uri_copy = group_read.uri()?;
+        assert_eq!(
+            group_uri_copy,
+            "file://".to_owned() + group_path.to_str().unwrap()
+        );
+        let group_type = group_read.query_type()?;
+        assert_eq!(group_type, QueryType::Read);
+
+        let open = group_read.is_open()?;
+        assert!(open);
+
+        std::mem::drop(group_read);
+        create_array(group_uri.to_owned() + "/aa", ArrayType::Dense)?;
+        create_array(group_uri.to_owned() + "/bb", ArrayType::Dense)?;
+        create_array(group_uri.to_owned() + "/cc", ArrayType::Sparse)?;
+
+        let mut group_write =
+            Group::open(&tdb, group_uri.to_owned(), QueryType::Write)?;
+        group_write.add_member("aa", true, Some("aa".to_owned()))?;
+        group_write.add_member("bb", true, Some("bb".to_owned()))?;
+        group_write.add_member("cc", true, Some("cc".to_owned()))?;
+        std::mem::drop(group_write);
+
+        let group_read =
+            Group::open(&tdb, group_uri.to_owned(), QueryType::Read)?;
+        let opt_string = group_read.dump(true)?;
+        match opt_string {
+            Some(s) => println!("{}", s),
+            None => println!("Empty group"),
+        }
+        std::mem::drop(group_read);
+
+        let mut group_write =
+            Group::open(&tdb, group_uri.to_owned(), QueryType::Write)?;
+        group_write.delete_member("bb")?;
+        std::mem::drop(group_write);
+
+        let group_read =
+            Group::open(&tdb, group_uri.to_owned(), QueryType::Read)?;
+        let opt_string = group_read.dump(true)?;
+        match opt_string {
+            Some(s) => println!("{}", s),
```

Comment:
Is there an assert that can be used to express what this wants instead?
If it's hard to check the contents of the string then asserting none/some seems fine.

For the `dump` API to be useful anyway it probably need to be parsed to some useful representation. If it's not documented then all anyone can do with it is print it or check some/none as you are doing here.